src/main/java/twilightforest/entity/boss/EntityTFMinoshroom.java

```diff
@@ -3,22 +3,33 @@
 import net.minecraft.entity.SharedMonsterAttributes;
 import net.minecraft.inventory.EntityEquipmentSlot;
 import net.minecraft.item.ItemStack;
+import net.minecraft.network.datasync.DataParameter;
+import net.minecraft.network.datasync.DataSerializers;
+import net.minecraft.network.datasync.EntityDataManager;
 import net.minecraft.util.DamageSource;
 import net.minecraft.util.ResourceLocation;
 import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.MathHelper;
 import net.minecraft.world.DifficultyInstance;
 import net.minecraft.world.EnumDifficulty;
 import net.minecraft.world.World;
+import net.minecraftforge.fml.relauncher.Side;
+import net.minecraftforge.fml.relauncher.SideOnly;
 import twilightforest.TFFeature;
 import twilightforest.TwilightForestMod;
 import twilightforest.block.BlockTFBossSpawner;
 import twilightforest.block.TFBlocks;
 import twilightforest.entity.EntityTFMinotaur;
+import twilightforest.entity.ai.EntityAITFGroundAttack;
 import twilightforest.enums.BossVariant;
 import twilightforest.item.TFItems;
 import twilightforest.world.TFWorld;
 
 public class EntityTFMinoshroom extends EntityTFMinotaur {
+	private static final DataParameter<Boolean> GROUND_ATTACK = EntityDataManager.<Boolean>createKey(EntityTFMinoshroom.class, DataSerializers.BOOLEAN);
+
+	private float clientSideChargeAnimation0;
```

Comment:
Not sure about the naming here, maybe follow mcp and use `prevClientSideChargeAnimation`? Other than that looks good to me